genrl/deep/agents/dqn/base.py

```diff
@@ -0,0 +1,164 @@
+from copy import deepcopy
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import gym
+import numpy as np
+import torch
+from torch import optim as opt
+from torch.nn import functional as F
+
+from ....environments import VecEnv
+from ...common import (
+    PushReplayBuffer,
+    get_env_properties,
+    get_model,
+    safe_mean,
+    set_seeds,
+)
+from ..base import BaseAgent
+
+
+class DQN(BaseAgent):
+    def __init__(
+        self,
+        network_type: str,
+        env: Union[gym.Env, VecEnv],
+        batch_size: int = 32,
+        gamma: float = 0.99,
+        layers: Tuple = (32, 32),
+        lr: float = 0.001,
```

Comment:
Should learning_rate be a part of the `BaseAgent`? Every agent needs it.

Reply:
Yeah. Will change it when I make the Base Off Policy Agent. There would be an `lr_policy` and `lr_value` in the Base Agent.